azurelinuxagent/common/version.py

```diff
@@ -51,19 +54,25 @@ def get_daemon_version():
     If the variable is not set (because the agent is < 2.2.53, or the process was not started by the daemon and
     the process is not the daemon itself) the function returns version of agent which started by the python
     """
+    # W0603: Using the global statement (global-statement)
+    # OK to disable; _DAEMON_VERSION is used only within get_daemon_version, but needs to persist across calls.
+    global _DAEMON_VERSION  # pylint: disable=W0603
     if __DAEMON_VERSION_ENV_VARIABLE in os.environ:
         return FlexibleVersion(os.environ[__DAEMON_VERSION_ENV_VARIABLE])
-    else:
+    elif _DAEMON_VERSION == FlexibleVersion("0.0.0.0"):
```

Comment:
rather than keeping this in a global, we should just set the environment variable, i.e. call set_daemon_version(version) and then return the env variable (or do a recursive call)